### PR TITLE
Try lld

### PR DESCRIPTION
**Pull request by matklad**

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ env:
   CARGO_NET_RETRY: 10
   CI: 1
   RUST_BACKTRACE: short
-  RUSTFLAGS: -D warnings
+  RUSTFLAGS: -D warnings -C linker=clang -C link-arg=-fuse-ld=lld
```
Comment (bjorn3) [inline .github/workflows/ci.yaml:15]: You may need to install lld first.

Comment (matklad) [reply .github/workflows/ci.yaml:15]: I kinda hoped that it is there by default, but no :(

My endg goal is to try lld on windows (as it is the CI bottleneck). I guess I can do this by seetting different RUSTFLAGS per OS, but I can't figure a convenient way to do that

**Comment by matklad** [comment]
Ok, got to work it on windows, but, apparently, it is slower? :-(

**Comment by matklad** [comment]
Hm, second try seems much faster, 

bors r+

**Comment by kjeremy** [comment]
Is that from the llvm-tools-preview component or whatever it's called? I actually thought those were stubs on windows.

**Comment by matklad** [comment]
I have no idea, tbh

**Comment by mati865** [comment]
> Is that from the llvm-tools-preview component or whatever it's called? I actually thought those were stubs on windows.

Rust-lld is always installed, `llvm-tools-preview` work fine (at least for windows-gnu).

For using system LLD on windows-gnu you use `-C link-arg=-fuse-ld=lld` and for bundled rust-lld `-C linker=rust-lld` on windows-msvc.
Using bundled rust-lld with windows-gnu/unix is blocked on https://github.com/rust-lang/rust/issues/71515

**Comment by matklad** [comment]
For posterity, I don't think this significantly improved overall compile time for windous on CI for us, so either the difference in linking time is small, or linking time is small relative to the rest of compilation. Note, however, that we cache all the dependencies, so linking time should be relatively more important than in the case of from-scratch compilation.

**Pull request merged**